api/handlers.go

```diff
@@ -89,12 +91,67 @@ func (api *API) unpinPOST(w http.ResponseWriter, req *http.Request, _ httprouter
 }
 
 // sweepPOST instructs pinner to scan the list of skylinks pinned by skyd and
-// update its database.
-func (api *API) sweepPOST(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
-	// TODO This should update both skylinks which are pinned but are not marked
-	//  as pinned by the local server and skylinks which are marked as pinned
-	//  but no longer are.
-	api.WriteError(w, errors.New("unimplemented"), http.StatusTeapot)
+// update its database. This is a very heavy call! Use with caution!
+func (api *API) sweepPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+
+	// Rebuild cache in a goroutine
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	var cacheErr error
+	go func() {
+		defer wg.Done()
+		cacheErr = api.staticSkydClient.RebuildCache()
+	}()
+
+	// Get pinned skylinks from the DB
+	dbSkylinks, err := api.staticDB.SkylinksForServer(req.Context(), api.staticServerName)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	wg.Wait()
+	if cacheErr != nil {
+		api.WriteError(w, cacheErr, http.StatusInternalServerError)
+		return
+	}
+
+	unknown, missing := api.staticSkydClient.DiffPinnedSkylinks(dbSkylinks)
+
+	ctx := req.Context()
```

Comment:
This is potentially a very slow operation. The context is almost guaranteed to time out.

Options:
1. Use a non-expiring background context.
2. Use a custom context with its own deadline which makes sense for this case.
3. Start the process in a separate goroutine (with a global semaphore which ensures only one sweep can be performed at a time) and respond with 202 Accepted. Optionally, we can provide a separate endpoint on which the status of the sweep can be monitored.

Reply:
From offline:

We want to make sure that callers cannot abuse this endpoint (on purpose or by accident), so we will limit this to a single execution at a time.

The best way to achieve this is by having this endpoint return immediately with a 202 Accepted and a response which points the caller to a `GET /sweep/status` endpoint which reports the status of the latest sweep - ongoing/finished.

F/U: Additionally, we want the status endpoint to return information about start and finish times for the sweep, number of unknown and missing skylinks handled, etc.